assets/docs/interactivity_tutorial.md

```diff
@@ -0,0 +1,441 @@
+* @param {String} [prefer] Sometimes, two sources provide a field with same name. Specify which
+* source will take precedence in the event of a conflict.
+*/
+function deNamespace(data, prefer) {
+    return Object.keys(data).reduce((acc, key) => {
+        const new_key = key.replace(/.*?:/, '');
+        if (!Object.prototype.hasOwnProperty.call(acc, new_key) || (!prefer || key.startsWith(prefer))) {
+            acc[new_key] = data[key];
+        }
+        return acc;
+    }, {});
+}
+```
+
+### Advanced alternative
+Sometimes, the existing page already has several widgets sharing data, and it would be difficult to rewrite things after the fact in a way that ceded control of data to LocusZoom. In this case, some compromise needs to be reached: how can LocusZoom fetch what it needs (possibly rendering only a subset of the data available), without duplicating API calls to the server that have already been made elsewhere?
+
+Reactive rendering frameworks sometimes solve the problem of sharing mutable data via a shared local cache (store) that represents data from the server. LocusZoom can then make requests to that store, and the store is then responsible for deciding whether a new server request is needed. This allows the result of a single API request to power all widgets on the page without redundant network traffic. The store call can then return a promise representing either local data, or the result of a server request, as appropriate.
+
+In this model, if the data changed due to an external event, LocusZoom might need to be notified that new data is available. This is best done using the standard methods used to trigger a re-render from outside the plot  (`plot.applyState()`).
+
+If you are not using a store, this manual re-render can also be triggered by watching an observable value. (many frameworks provide a way to manually watch values for changes ([eg vue.js](https://vuejs.org/v2/guide/computed.html#Watchers)).
+
+## Filters control what is shown
+Filters can be used to control what elements are shown on a panel. This will hide elements, but preserve the space that those elements would have occupied: eg, the axis limits will reflect all of the data, not just what is currently shown.
+
+Filters are specified in data layout layouts, and can be applied to most data layer types.
+```javascript
+{
+  ...options,
+  filters: [
+    { field: '{{namespace[access]}}score', operator: '!=', value: null },
+  ],
+}
+```
+
+Scatter plots have an additional option to show labels for some or all data elements. This is controlled via adding a similar filter block inside *label*:
+```javascript
+{
+  id: 'phewaspvalues',
+  ...options,
+  label: {
+    text: '{{{{namespace[phewas]}}trait_label}}',
+    filters: [
+      {
+        field: '{{namespace[phewas]}}log_pvalue',
+        operator: '>=',
+        value: 20,
+      },
+    ],
+  }
+}
+```
+
+The following filters are available:
+
+* `=`: field value *exactly* matches the provided value
```

Comment:
TODO: #207 will refactor filters to use a registry (users can inject their own logic and custom operators, fetched by name from the registry)